modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
 		forge_temperature += 5
 
 	if(forge_temperature > 0)
-		if(primitive)
-			icon_state = "primitive_forge_full"
-		else
-			icon_state = "forge_full"
+		icon_state = "[primitive ? "primitive_" : ""]forge_full"
 		light_range = 3
 	else if(forge_temperature <= 0)
-		if(primitive)
-			icon_state = "primitive_forge_empty"
-		else
-			icon_state = "forge_empty"
+		icon_state = "[primitive ? "primitive_" : ""]forge_empty"
 		light_range = 0
 
+/**
+ * Here we fix any weird in_use bugs
+ */
 /obj/structure/reagent_forge/proc/check_in_use()
 	if(!in_use)
 		return
 	for(var/mob/living/living_mob in range(1,src))
 		if(!living_mob)
 			in_use = FALSE
 
+/**
+ * Here we spawn coal depending on a chance of using wood
+ */
 /obj/structure/reagent_forge/proc/spawn_coal()
 	new /obj/item/stack/sheet/mineral/coal(get_turf(src))
```

Comment:
This proc feels kinda useless if its just a single line anyways

Reply:
also, perhaps you can set the name to be charcoal, simply for flavor reasons? 👀 should be possible to set a custom name uh... somehow. I might do that one myself later on though, so low-priority

Reply:
yea, but its for a callback